src/components/network/network-renderer.js

```diff
@@ -565,3 +567,122 @@ genotet.NetworkRenderer.prototype.findSelectEdges = function(edgeIds) {
     this.signal('showMultiEdges');
   }
 };
+
+/**
+ * Initializes the polygon selection mode.
+ * @private
+ */
+genotet.NetworkRenderer.prototype.polygonSelectionInit_ = function() {
+  var coords = [];
+  var line = d3.svg.line();
+  var g = this.canvas.append('g');
```

Comment:
You might be appending a new g each time. Give it a class and remove it precisely. Do not selectALl and remove.